shap/utils/_clustering.py

```diff
@@ -116,8 +118,8 @@ def xgboost_distances_r2(
     X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, y, random_state=random_state)
 
     # fit an XGBoost model on each of the features
-    test_preds = []
-    train_preds = []
+    test_preds_list = []
+    train_preds_list = []
```

Comment:
Very very minor nitpick, but could we shift the initialization of `train_preds_list` on top of `test_preds_list`?